src/Bicep.RegistryModuleTool/ModuleFiles/MainArmTemplateFile.cs

```diff
@@ -98,36 +95,55 @@ public MainArmTemplateFile WriteToFileSystem(IFileSystem fileSystem)
             return this;
         }
 
-        private static MainArmTemplateParameter ToParameter(JsonProperty parameterProperty)
+        private MainArmTemplateParameter ToParameter(JsonProperty parameterProperty)
         {
             string name = parameterProperty.Name;
-            string type = parameterProperty.Value.GetProperty("type").ToNonNullString();
+            string type = GetTypeFromDefinition(parameterProperty.Value);
             bool required = !parameterProperty.Value.TryGetProperty("defaultValue", out _);
             string? description = TryGetDescription(parameterProperty.Value);
 
             return new(name, type, required, description);
         }
 
-        private static MainArmTemplateOutput ToOutput(JsonProperty outputProperty)
+        private MainArmTemplateOutput ToOutput(JsonProperty outputProperty)
         {
             string name = outputProperty.Name;
-            string type = outputProperty.Value.GetProperty("type").ToNonNullString();
+            string type = GetTypeFromDefinition(outputProperty.Value);
             string? description = TryGetDescription(outputProperty.Value);
 
             return new(name, type, description);
         }
 
-        private static string? TryGetDescription(JsonElement element)
+        private string GetTypeFromDefinition(JsonElement element)
```

Comment:
<img width="883" alt="image" src="https://user-images.githubusercontent.com/9027725/231220095-a2ea261e-3536-4038-bcf0-3a64ae9ec62d.png">

This may be trivial, just need to figure out the type conversions (but I may need help :-D)

Reply:
all done!